### PR TITLE
Webflow fixes

### PR DESCRIPTION
**Pull request by awan7676**
1. Fetch Users from all the sites
2. Delete user from all the sites

### CODE REVIEW

File: apps/webflow/src/inngest/functions/users/sync-users-page.ts [hunk 1]
```diff
@@ -62,39 +60,53 @@ export const syncUsersPage = inngest.createFunction(
...
-        name: 'webflow/users.page_sync.requested',
-        data: {
-          ...event.data,
-          page: nextPage,
-        },
-      });
-      return {
-        status: 'ongoing',
-      };
+      while (nextPage !== null) {
```
Comment (athielba) [inline apps/webflow/src/inngest/functions/users/sync-users-page.ts:71]: You should not handle the pagination like this, you need to use `waitForEvent` event, first you have to complete all the user listing for a single site, & wait for all the  task complete, then call the second site , check this  [example](https://github.com/elba-security/elba-security/blob/staging/apps/dropbox/src/inngest/functions/data-protection/start-shared-links-sync.ts#L59)

File: apps/webflow/src/app/oauth/service.ts [hunk 2]
```diff
@@ -15,24 +17,25 @@ export const setupOrganisation = async ({
...
+    throw new Error('No sites found for the organisation');
   }
   await db
     .insert(Organisation)
     .values({
       id: organisationId,
-      accessToken,
-      siteId,
+      accessToken: encodedAccessToken,
+      siteIds,
```
Comment (athielba) [inline apps/webflow/src/app/oauth/service.ts:30]: Site ids are not static,  an account  admin may create sites continuously, therefore it should be fetch  when the sync user page event runs, if you have any concerns let me know

File: apps/webflow/src/inngest/functions/users/delete-user.ts [hunk 3]
```diff
@@ -36,8 +39,12 @@ export const deleteWebflowUser = inngest.createFunction(
       return result;
     });
 
-    await step.run('delete-user', async () => {
-      await deleteUser(organisation.accessToken, organisation.siteId, id);
-    });
+    const token = await decrypt(organisation.accessToken);
+
+    for (const siteId of organisation.siteIds){
```
Comment (athielba) [inline apps/webflow/src/inngest/functions/users/delete-user.ts:44]: steps should be in the `Promise.all()`, check this [example](https://github.com/elba-security/elba-security/pull/281/commits/ee243fc97792324517a95ef51b9abd2ace1a0a5e#diff-c20e7012dca634037c64e88f6bc4ae27f13950727b23be4b7f32b177d08e3b0cL51)

File: apps/webflow/src/app/api/webhooks/elba/users/delete/service.ts [hunk 4]
```diff
@@ -1,19 +1,17 @@
...
-    userIds.map((userId) => ({
-      name: 'webflow/users.delete.requested',
-      data: {
-        id: userId,
-        organisationId,
-      },
-    }))
-  );
-};
+  await inngest.send({
```
Comment (athielba) [inline apps/webflow/src/app/api/webhooks/elba/users/delete/service.ts:10]: you should not update this, please revert it (1 user should be handled in the `calendly/users.delete.requested` event), previous implementation is the correct way to do

File: apps/webflow/src/app/oauth/service.test.ts [hunk 5]
```diff
@@ -84,28 +83,29 @@ describe('setupOrganisation', () => {
...
+    }
+    expect(storedOrganisation.region).toBe(region);
+    await expect(decrypt(storedOrganisation.accessToken)).resolves.toEqual(accessToken);
+
 
     // verify that the user/sync event is sent
     expect(send).toBeCalledTimes(1);
     expect(send).toBeCalledWith({
-      name: 'webflow/users.page_sync.requested',
+      name: 'webflow/users.sync.requested',
```
Comment (athielba) [inline apps/webflow/src/app/oauth/service.test.ts:104]: It is only updated here, What about other places ?

File: apps/webflow/src/inngest/functions/users/start-users-sync.ts [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+      },
+      retries: 5,
+      cancelOn: [
+        {
+          event: 'webflow/app.uninstall.requested',
+          match: 'data.organisationId',
+        },
+      ],
+    },
+    { event: 'webflow/users.sync.requested' },
```
Comment (athielba) [inline apps/webflow/src/inngest/functions/users/start-users-sync.ts:26]: the event should be `webflow/users.start_sync.requested'`

File: apps/webflow/src/inngest/functions/users/start-users-sync.ts [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+import { db } from '@/database/client';
+import { Organisation } from '@/database/schema';
+import { inngest } from '@/inngest/client';
+import { createElbaClient } from '@/connectors/elba/client';
+import { getSiteIds } from '@/connectors/webflow/sites';
+import { decrypt } from '@/common/crypto';
+
+export const syncUsers = inngest.createFunction(
+    {
+      id: 'webflow-sync-users',
```
Comment (athielba) [inline apps/webflow/src/inngest/functions/users/start-users-sync.ts:13]: `webflow-sync-users` -> webflow-start-sync-users

File: apps/webflow/src/inngest/functions/users/start-users-sync.ts [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+
+      const siteIds = await step.run('get-site-ids', async () => {
+        const result = await getSiteIds(token);
+        return result
+      });
+  
+      // Process each site one by one
+      for (const siteId of siteIds) {
+        await step.sendEvent('sync-users-page', {
+          name: 'webflow/users.page_sync.requested',
```
Comment (athielba) [inline apps/webflow/src/inngest/functions/users/start-users-sync.ts:56]: it should call  'webflow/users.sync.requested'

File: apps/webflow/src/inngest/types.ts [hunk 9]
```diff
@@ -2,18 +2,30 @@ import { type GetFunctionInput } from 'inngest';
...
     data: {
       organisationId: string;
       region: string;
-      isFirstSync: boolean;
-      syncStartedAt: number;
       page: number;
+      siteId: string;
+    };
+  };
+  'webflow/users.site_sync.completed': {
```
Comment (athielba) [inline apps/webflow/src/inngest/types.ts:20]: `webflow/users.sync.completed`

File: apps/webflow/src/inngest/functions/users/sync-users-page.ts [hunk 10]
```diff
@@ -80,7 +75,6 @@ export const syncUsersPage = inngest.createFunction(
       return null;
     });
 
-    // if there is a next page enqueue a new sync user event
     if (nextPage) {
       await step.sendEvent('sync-users-page', {
         name: 'webflow/users.page_sync.requested',
```
Comment (athielba) [inline apps/webflow/src/inngest/functions/users/sync-users-page.ts:80]:
```suggestion
        name: 'webflow/users.sync.requested',
```

**Comment by awan7676** [comment]
@athielba working on these comments

File: apps/webflow/src/inngest/functions/users/delete-user.ts [hunk 11]
```diff
@@ -36,8 +39,17 @@ export const deleteWebflowUser = inngest.createFunction(
...
+    const token = await decrypt(organisation.accessToken);
+
+    const siteIds = await step.run('get-site-ids', async () => {
+      const result = await getSiteIds(token);
+      return result
     });
+
+    for (const siteId of siteIds){
+      await step.run('delete-user', async () => {
+        await Promise.all(ids.map((id) => deleteUser(token, siteId, id)));
```
[the remark this answers is not in the document]
Comment (awan7676) [reply apps/webflow/src/inngest/functions/users/delete-user.ts:51]: Updated this according to the zoom PR

File: apps/webflow/src/app/api/webhooks/elba/users/delete/service.ts (hunk 4 above)
Comment (awan7676) [reply apps/webflow/src/app/api/webhooks/elba/users/delete/service.ts:10]: Done

File: apps/webflow/src/inngest/functions/users/start-users-sync.ts (hunk 6 above)
Comment (awan7676) [reply apps/webflow/src/inngest/functions/users/start-users-sync.ts:26]: changes it in all files

File: apps/webflow/src/inngest/functions/users/start-users-sync.ts (hunk 7 above)
Comment (awan7676) [reply apps/webflow/src/inngest/functions/users/start-users-sync.ts:13]: Done

File: apps/webflow/src/inngest/functions/users/start-users-sync.ts (hunk 8 above)
Comment (awan7676) [reply apps/webflow/src/inngest/functions/users/start-users-sync.ts:56]: for each site is, we are calling 'webflow/users.page_sync.requested' which basically syncs all the users for that particular siteId. 'webflow/users.sync.requested' is for starting the sync with all the siteIds. Isn't this implementation correct?

File: apps/webflow/src/inngest/types.ts (hunk 9 above)
Comment (awan7676) [reply apps/webflow/src/inngest/types.ts:20]: Done

File: apps/webflow/src/inngest/functions/users/sync-users-page.ts (hunk 10 above)
Comment (awan7676) [reply apps/webflow/src/inngest/functions/users/sync-users-page.ts:80]: Done

File: apps/webflow/src/app/oauth/service.test.ts (hunk 5 above)
Comment (awan7676) [reply apps/webflow/src/app/oauth/service.test.ts:104]: Updated it in all files

File: apps/webflow/src/inngest/functions/users/start-users-sync.ts (hunk 8 above)
Comment (athielba) [reply apps/webflow/src/inngest/functions/users/start-users-sync.ts]: You should have 2 events (please make sure to keep proper naming for each evens)
1. 'webflow/users.start_sync.requested is responsible to start the sync
2. webflow/users.sync.requested is responsible for sync the users

`webflow/users.page_sync.requested` should be renamed to `webflow/users.sync.requested`, I am not sure what is the issue here

Comment (awan7676) [reply apps/webflow/src/inngest/functions/users/start-users-sync.ts]: Yeah It's done

**Pull request merged**